git_data_model.sql

```diff
@@ -1,160 +1,337 @@
+VALUES ('ISSUE1', 2,'resolved','feature_request',1);
 
 
 SELECT * FROM issues;
 
+CREATE TABLE issue_comments (
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  issue_id BIGINT NOT NULL,
+  user_id BIGINT NOT NULL,
+  created_at TIMESTAMP WITHOUT TIME ZONE DEFAULT(NOW() AT TIME ZONE 'UTC'),
+  comments TEXT NOT NULL,
+  FOREIGN KEY (issue_id) REFERENCES issues(id),
+  FOREIGN KEY (user_id) REFERENCES users(id)
+);
+
+INSERT INTO issue_comments (issue_id,user_id,comments)
+VALUES (1,1,'the button is not working');
+
+SELECT * FROM issue_comments;
+
+
 CREATE TABLE pull_requests (
-  pull_request_id BIGSERIAL PRIMARY KEY,
-  title VARCHAR (255),
-  descriptions VARCHAR (255) NOT NULL,
-  author BIGINT ,
-  source_branch BIGINT ,
-  target_branch BIGINT,
-  commit_id VARCHAR,
-  statuses  VARCHAR,
-  dates DATE,
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  title VARCHAR (255) NOT NULL,
+  description VARCHAR (255) NOT NULL,
+  user_id BIGINT NOT NULL,
+  created_at TIMESTAMP WITHOUT TIME ZONE DEFAULT(NOW() AT TIME ZONE 'UTC'),
+  updated_at TIMESTAMP WITHOUT TIME ZONE DEFAULT(NOW() AT TIME ZONE 'UTC'),
+  source_branch BIGINT NOT NULL,
+  target_branch BIGINT NOT NULL,
+  status  VARCHAR NOT NULL,
   comments TEXT,
-  repo_id BIGINT,
-  UNIQUE (pull_request_id),
-  FOREIGN KEY (author) REFERENCES users(user_id),
-  FOREIGN KEY (source_branch) REFERENCES branches(branch_id),
-  FOREIGN KEY (target_branch) REFERENCES branches(branch_id),
-  FOREIGN KEY (commit_id) REFERENCES commits(commit_hash),
-  FOREIGN KEY (repo_id) REFERENCES repo(repo_id)
+  repo_id BIGINT NOT NULL,
+  FOREIGN KEY (user_id) REFERENCES users(id),
+  FOREIGN KEY (source_branch) REFERENCES branches(id),
+  FOREIGN KEY (target_branch) REFERENCES branches(id),
+  FOREIGN KEY (repo_id) REFERENCES repos(id)
 );
 
 
-INSERT INTO pull_requests (title,descriptions,author,source_branch,target_branch,commit_id,statuses,dates,comments,repo_id)
-VALUES ('Ipull1','desc1', 2,2,1,'abcd','unmerged', '2022-01-01','worked on the styles',1);
+INSERT INTO pull_requests (title,description,user_id,source_branch,target_branch,status,comments,repo_id)
+VALUES ('pull1','desc1', 2,2,1,'unmerged','worked on the styles',1);
+
+INSERT INTO pull_requests (title,description,user_id,source_branch,target_branch,status,comments,repo_id)
+VALUES ('pull2','desc2', 1,2,1,'raised','worked on the bug',1);
+
+INSERT INTO pull_requests (title,description,user_id,source_branch,target_branch,status,comments,repo_id)
+VALUES ('pull3','desc3', 2,2,1,'rejected','worked on the feature pop up',1);
+
+INSERT INTO pull_requests (title,description,user_id,source_branch,target_branch,status,comments,repo_id)
+VALUES ('pull4','desc4', 2,2,1,'merged','refactored the code',1);
+
+INSERT INTO pull_requests (title,description,user_id,source_branch,target_branch,status,comments,repo_id)
+VALUES ('pull5','desc5', 2,2,1,'merged','refactored the code again',1);
 
 
 SELECT * FROM pull_requests;
 
+CREATE TABLE pull_request_commits (
```

Comment:
this commits any different from commits you are storing in `commits` table? Why is it required to store them separately per PR?

Reply:
sir i dit it to store the commits included in a pull request. if we store it two separate tables then when we only need the pull request data it could have better performance when we have larger dataset and the pull request would have more querying .That was the thought behind it.